app/models/state_file_ny_intake.rb

```diff
@@ -260,10 +260,23 @@ def disqualifying_eligibility_rules
     }
   end
 
-  def self.locality_nm_valid?(locality)
-    (LOCALITIES.detect do |loc|
-      locality.starts_with?(loc)
-    end).present?
+  def self.locality_nm_valid?(locality_nm)
+    locality_nm = locality_nm.upcase
+    (LOCALITIES.detect { |loc| locality_nm.starts_with?(loc) }).present?
+  end
+
+  def validate_w2(w2)
```

Comment:
this feels like a bit of a misnomer because it's not really telling you if the whole w2 is valid, just the locality name

Reply:
oh ok i get it now. maybe consider naming it `validate_state_specific_w2_requirements` or something like that? only suggesting it because it's generally pretty confusing that we have 2 sets of validations for DF W2s and our model W2s and within our model W2s we have validations that live on that model and also these ones that live on the intake model. so anything we can do to add more context than just "validate" and "w2" is maybe good

Reply:
Good call - done.